### ISSUE TITLE
[Bug]: Header bar looks broken on mobile on site's frontend

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#57352]
### Quick summary

On some themes (Quadrat, FSE beta experience), mobile header looks broken:

<img width="365" alt="Screenshot 2021-10-26 at 14 32 54" src="https://user-images.githubusercontent.com/87168/138870583-1fe9c68d-c06c-488c-8ac9-85b8172f1b0c.png">

After 480px+ it looks ok again:

<img width="602" alt="Screenshot 2021-10-26 at 14 33 32" src="https://user-images.githubusercontent.com/87168/138870643-49b20f27-fc10-411e-b317-97b5553fc223.png">

Site ID 199113916.

On Atomic blocks theme (On Atomic site mikaelkorpela.fi):

<img width="601" alt="Screenshot 2021-10-26 at 14 33 59" src="https://user-images.githubusercontent.com/87168/138870548-1befb712-2530-4dec-af46-5f8f062a2144.png">

On Hever header works:

<img width="316" alt="Screenshot 2021-10-26 at 14 38 56" src="https://user-images.githubusercontent.com/87168/138870486-34cc5cad-c067-4ce0-9478-c38076e72cb3.png">

Could be some themes leak CSS into the header and it needs some CSS hardening?

### Steps to reproduce

- Use Firefox
- Stay logged in at WP.com
- Open above site's homepage
- Try mobile screen width

### What you expected to happen

Functioning header

### What actually happened

Header buttons float out of the header.

### Browser

Firefox.

### Simple/Atomic

Both.

### Reproducibility

Consistent

**Comment by sixhours** [comment]
I can reproduce this on my test site `calobeetesting44.wordpress.com` but only after activating FSE, and only with some themes.

I spot-checked diffs for Quadrat and Blockbase back to late September and saw no improvements when reverting them, so I don't think this is directly related to theme changes.

Going to check on the FSE beta angle next.

**Comment by sixhours** [comment]
Looks #57167 was also reported. Closing that as a dupe.

**Comment by sixhours** [comment]
This is happening because a new "Edit site" link appears in the header bar for FSE beta sites that have FSE-enabled themes (like Quadrat, not Hever), but isn't accounted for or removed on small screens. Demo in the inspector:

https://user-images.githubusercontent.com/2124984/139083718-76de8dfe-dd42-439f-b9a1-39c94ee86993.mov

Will ping Cylon and get their eyes on this.

**Comment by sixhours** [comment]
The weirdness with the avatar on Atomic sites is unrelated to this issue and only happens because we're logged in and proxied A11ns viewing our own sites. We can ignore that for now.

**Comment by sixhours** [comment]
Confirmed with FSE folks in Slack that the menu item comes from Gutenberg:

https://github.com/WordPress/gutenberg/blob/trunk/lib/full-site-editing/full-site-editing.php#L69-L99

I've confirmed that the issue doesn't happen with Gutenberg on a core WP install; the link gets `display: none`ed on mobile devices. It's likely our styles on WP.com override the styles for this link:

<img width="1511" alt="Screen Shot 2021-10-27 at 12 22 34 PM" src="https://user-images.githubusercontent.com/2124984/139106738-8830d588-8618-4ce4-a362-984dfa81c3d0.png">

I'm not sure when it was activated on WP.com, but we can throw up a patch on the WP.com side to hide the button on small screens.

**Comment by sixhours** [comment]
Added a quick fix here: D69139-code

**Comment by sixhours** [comment]
Out of curiosity, I tested the top bar on my WP.com-connected, non-Atomic Jetpack site and, similar to Atomic sites for proxied A11ns, it looks totally broken:

<img width="710" alt="Screen Shot 2021-10-27 at 4 58 13 PM" src="https://user-images.githubusercontent.com/2124984/139146069-bdcc1a1a-fa5b-4621-8f94-6b077e56d586.png">

All the buttons save for the avatar disappear on tablet-size screens. :-/ So there's definitely a larger issue at play, beyond the scope of this fix. I'll open a separate issue.

**Comment by sixhours** [comment]
Patched in r234201-wpcom
